Makefile.js

```diff
@@ -865,6 +865,10 @@ target.checkRuleFiles = function() {
 
 };
 
+target.checkRuleExamples = function() {
+    exec(`${NODE}tools/check-rule-examples.js docs/src/rules/*.md`);
```

Comment:
This doesn't seem to work on Windows, probably because [`exec()`](https://github.com/shelljs/shelljs#execcommand--options--callback) runs commands through `cmd.exe` which doesn't expand `*.md`.

```
$ node Makefile.js checkRuleExamples

docs/src/rules/*.md
  0:0  error  Error checking file: ENOENT: no such file or directory, open 'C:\projects\eslint\docs\src\rules\*.md'

✖ 1 problem (1 error, 0 warnings)

C:\projects\eslint\node_modules\shelljs\src\common.js:401
      if (config.fatal) throw e;
                        ^

Error: exec:
docs/src/rules/*.md
  0:0  error  Error checking file: ENOENT: no such file or directory, open 'C:/projects/eslint/docs/src/rules/*.md'

✖ 1 problem (1 error, 0 warnings)


    at Object.error (C:\projects\eslint\node_modules\shelljs\src\common.js:110:27)
    at execSync (C:\projects\eslint\node_modules\shelljs\src\exec.js:120:12)
    at _exec (C:\projects\eslint\node_modules\shelljs\src\exec.js:223:12)
    at C:\projects\eslint\node_modules\shelljs\src\common.js:335:23
    at target.checkRuleExamples (C:\projects\eslint\Makefile.js:869:5)
    at global.target.<computed> [as checkRuleExamples] (C:\projects\eslint\node_modules\shelljs\make.js:36:40)
    at C:\projects\eslint\node_modules\shelljs\make.js:48:27
    at Array.forEach (<anonymous>)
    at Timeout._onTimeout (C:\projects\eslint\node_modules\shelljs\make.js:46:10)
    at listOnTimeout (node:internal/timers:573:17)

Node.js v20.7.0
```

Perhaps we could pass no arguments and in that case find all rule docs files in the tool? Something like `fetch-docs-links` is doing:

https://github.com/eslint/eslint/blob/21ebf8a811be9f4b009cf70a10be5062d4fdc736/tools/fetch-docs-links.js#L43-L48

Reply:
I tested this on GithHub Actions with windows-latest and it seemed to be working, so it's possible that I did a mistake, or maybe the containers have a special configuration that would let the task run. Now I'm testing on a real Windows machine, and I can reproduce the error above.

But I noticed that there is also another glitch: when shelljs' `exec` is used to run the command, color and underlining are not rendered in the terminal.

This is how it looks like when I run the script directly with node:

![image](https://github.com/eslint/eslint/assets/6367844/26a9f863-1962-44ae-b67b-abd0034e23bf)

And this is what I'm getting when I run it with `exec` in the Makefile:

![image](https://github.com/eslint/eslint/assets/6367844/40416fb8-b142-4e6b-bc16-29a5bd819241)

Probably because `stout` is not a TTY stream in the second example. It would be nice to fix that as well. I'll see what I can do.

Reply:
I fixed the problem by using glob to expand the patterns passed on the command line into file names, and I stopped using `exec` more because its behavior depends on the system shell and is not the consistent on Windows and Unix. I'm using now Node's built-in `execFileSync` which correctly renders colors and underlining in the terminal.

I've also updated the unit tests to ensure that the sequences for colors and underlining are inserted in the output (only for `check-rule-example.js`, because there are no tests for the Makefile). We could also revert this change if it makes the output in the tests difficult to read.